packages/nextjs/src/config/wrappers/wrapperUtils.ts

```diff
@@ -38,12 +45,15 @@ export function withErrorInstrumentation<F extends (...args: any[]) => any>(
   return function (this: unknown, ...origFunctionArguments: Parameters<F>): ReturnType<F> {
     try {
       const potentialPromiseResult = origFunction.call(this, ...origFunctionArguments);
+
       // First of all, we need to capture promise rejections so we have the following check, as well as the try-catch block.
       // Additionally, we do the following instead of `await`-ing so we do not change the method signature of the passed function from `() => unknown` to `() => Promise<unknown>.
       Promise.resolve(potentialPromiseResult).catch(err => {
         // TODO: Extract error logic from `withSentry` in here or create a new wrapper with said logic or something like that.
         captureException(err);
+        throw err;
```

Comment:
Small bugfix: we need to rethrow the rejected promise here, otherwise it counts as "caught"

Reply:
Okay so this turned out to be problematic. Throwing here caused an uncaught promise rejection to bubble up, which terminates the process in Node 16 and above. I think my approach here was overly clever from the beginning so I changed it to simply await the maybe-promise result. This makes the wrapped function async but as far as I can tell that is fine since everywhere we use it, we're already async.

Reply:
Thinking more about this, if we really want this wrapper function not to be changing the function signature in the future, we can go with that approach, but without the `throw`.

I think the behaviour stays as expected: `await`ing a rejected Promise throws, and you can also catch it via `.catch()`. The only drawback is, that it will be marked as "caught", not causing an unhandled promise rejection to bubble up. We can maybe circumvent this via `queueMicrotask`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises#when_promises_and_tasks_collide. But that again sounds a bit overly clever and is complicated to understand after the fact.